File: wallet/src/main/java/com/worldventures/wallet/ui/settings/help/video/adapter/VideoLocaleAdapter.java
package com.worldventures.wallet.ui.settings.help.video.adapter;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.worldventures.core.modules.video.model.VideoLocale;
import com.worldventures.wallet.R;

import java.util.List;

public class VideoLocaleAdapter extends ArrayAdapter<VideoLocale> {

   private final LayoutInflater layoutInflater;
   private final List<VideoLocale> data;

   public VideoLocaleAdapter(@NonNull Context context, List<VideoLocale> data) {
      super(context, R.layout.item_wallet_video_locale, data);
      layoutInflater = LayoutInflater.from(context);
      this.data = data;
   }

   @Override
   public View getDropDownView(int position, @Nullable View view, @NonNull ViewGroup parent) {
      ViewHolder holder;
      if (view == null) {
         view = layoutInflater.inflate(R.layout.item_wallet_video_locale, parent, false);

         holder = new ViewHolder();
         holder.tvName = view.findViewById(R.id.tv_name);
         holder.ivFlag = view.findViewById(R.id.iv_flag);

         view.setTag(holder);
      } else {
         holder = (ViewHolder) view.getTag();
      }

      final VideoLocale videoLocale = data.get(position);

      holder.tvName.setTextColor(Color.BLACK);
      holder.tvName.setText(videoLocale.getTitle());
      holder.ivFlag.setImageURI(videoLocale.getImage());

      return view;
   }

   @NonNull
   @Override
   public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
      View view = getDropDownView(position, convertView, parent);
      ((ViewHolder) view.getTag()).tvName.setTextColor(Color.WHITE);
      return view;
   }

   private static class ViewHolder {
      TextView tvName;
      SimpleDraweeView ivFlag;
   }
}

File: wallet/src/main/java/com/worldventures/wallet/ui/settings/help/video/impl/language/adapter/VideoLocaleAdapter.kt
package com.worldventures.wallet.ui.settings.help.video.impl.language.adapter

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.facebook.drawee.view.SimpleDraweeView
import com.worldventures.core.modules.video.model.VideoLocale
import com.worldventures.wallet.R

class VideoLocaleAdapter(context: Context, private val data: List<VideoLocale>) : ArrayAdapter<VideoLocale>(context, 0, data) {

   private val layoutInflater: LayoutInflater = LayoutInflater.from(context)

   override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
      val (view, holder) = prepareHolder(convertView, parent)
      val videoLocale = data[position]

      holder.tvName.setTextColor(Color.BLACK)
      holder.tvName.text = videoLocale.title
      holder.ivFlag.setImageURI(videoLocale.image)
      return view
   }

   private fun prepareHolder(view: View?, parent: ViewGroup): Pair<View, ViewHolder> {
      return if (view == null) {
         val currentView = layoutInflater.inflate(R.layout.item_wallet_video_locale, parent, false)
         val holder = ViewHolder(
               currentView.findViewById(R.id.tv_name),
               currentView.findViewById(R.id.iv_flag))
         currentView.tag = holder
         Pair(currentView, holder)
      } else {
         Pair(view, view.fetchViewHolder())
      }
   }

   private class ViewHolder(
         val tvName: TextView,
         val ivFlag: SimpleDraweeView
   )

   @Suppress("UnsafeCast")
   private fun View.fetchViewHolder(): ViewHolder = tag as ViewHolder
}
